commands/team.go

```diff
@@ -319,3 +336,80 @@ func removeDuplicatesAndSortTeams(teams []*model.Team) []*model.Team {
 	return result
 }
+
+func renameTeamCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 || args[0] == "" {
+		return errors.New("Error: requires at least 1 arg(s), only received 0")
+	}
+
+	oldTeamName := args[0]
+	newDisplayName, _ := cmd.Flags().GetString("display_name")
+	newTeamName, _ := cmd.Flags().GetString("name")
+
+	// If both flags are absent, abort!
+	if newTeamName == "" && newDisplayName == "" {
+		return errors.New("Require atleast one flag to rename team, either 'name' or 'display_name'")
+	}
+
+	team := getTeamFromTeamArg(c, oldTeamName)
+	if team == nil {
+		return errors.New("Unable to find team '" + oldTeamName + "', to see the all teams try 'team list' command")
+	}
+
+	// Check for name uniqueness
+	if newTeamName != "" && newTeamName == team.Name {
+		return errors.New("Entered name is the current name for " + team.Name + " , either remove the flag or updage to new value")
+	}
+
+	// Check for display name uniqueness
+	if newDisplayName != "" && newDisplayName == team.DisplayName {
+		return errors.New("Entered display name is the current display name for " + team.Name + " , either remove the flag or updage to new value")
+	}
+
+	if newTeamName != "" {
+		team.Name = newTeamName
+	}
+	if newDisplayName != "" {
+		team.DisplayName = newDisplayName
+	}
+
+	// Using updateTeam to rename team
+	updatedTeam, response := c.UpdateTeam(team)
+	if response.Error != nil {
+		return errors.New("Cannot rename team '" + oldTeamName + "', error : " + response.Error.Error())
+	}
+
+	// Checking if update from API did actually occur
+	isTeamNameEmpty := newTeamName == ""
+	isDisplayNameEmpty := newDisplayName == ""
+	wasTeamNameUpdated := newTeamName == updatedTeam.Name
+	wasDisplayNameUpdated := newDisplayName == updatedTeam.DisplayName
+
+	if isTeamNameEmpty && wasDisplayNameUpdated {
+		printer.Print("Successfully renamed team '" + oldTeamName + "'")
+		return nil
+	}
+	if isTeamNameEmpty && !wasDisplayNameUpdated {
+		return errors.New("Failed to update display name of team '" + oldTeamName + "'")
+	}
+
+	if isDisplayNameEmpty && wasTeamNameUpdated {
+		printer.Print("Successfully renamed team '" + oldTeamName + "'")
+		return nil
+	}
+	if isDisplayNameEmpty && !wasTeamNameUpdated {
+		return errors.New("Failed to update name of team '" + oldTeamName + "'")
+	}
+
+	if !isTeamNameEmpty && !isDisplayNameEmpty && wasTeamNameUpdated && wasDisplayNameUpdated {
+		printer.Print("Successfully renamed team '" + oldTeamName + "'")
+		return nil
+	}
+	if !isTeamNameEmpty && !isDisplayNameEmpty && !wasTeamNameUpdated && wasDisplayNameUpdated {
+		return errors.New("Partially successfull, could not update name of team '" + oldTeamName + "'")
+	}
+	if !isTeamNameEmpty && !isDisplayNameEmpty && wasTeamNameUpdated && !wasDisplayNameUpdated {
+		return errors.New("Partially successfull, could not update display name of team '" + oldTeamName + "'")
+	}
+	return errors.New("Failed to rename team '" + oldTeamName + "'")
```

Comment:
These checks are not necessary, we should trust the API return value, as it would return an error if the changes were not applied. Having an OK from the API with the changes not applied would be a bug in the API itself